# Machine Learning

Input  Output     formula
2        6        2**3 -2
3        24       3**3 - 3
4        60       4**3 - 4 
5        120       5**3 - 5

formula
model


Traditional method
input 
formula

x=5   y=x**3 - x 

In [8]:
import pandas as pd

In [10]:
data = pd.read_csv("C:/Users/user/Desktop/learn_data/diabetes.csv") 
data.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [12]:
data

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

#to predict diabetes

Machine learning types
1. Supervised learning   (input and output data ma deko cha bhane and output herera result diney garney), also called labeled data

Types  
- Regression  x,y (output, target value):  numerical  example: numeric output huncha
- Classification y() : class label     example: Titanic dataset
Algorithms
- Linear regression
 
2. Unsupervised learning ( no label data and target value is given, check the pattern and cluster form garcha)
3. Reinforcement learning

# Linear Regression Algorithm

In [18]:
data = pd.read_csv("C:/Users/user/Desktop/learn_data/insurance.csv") 
data.head()

age     sex   bmi  children smoker     region  expenses
0   19  female  27.9         0    yes  southwest  16884.92
1   18    male  33.8         1     no  southeast   1725.55
2   28    male  33.0         3     no  southeast   4449.46
3   33    male  22.7         0     no  northwest  21984.47
4   32    male  28.9         0     no  northwest   3866.86

In [24]:
#age -- expenses

df=data[['age','expenses']]
df.head()

age  expenses
0   19  16884.92
1   18   1725.55
2   28   4449.46
3   33  21984.47
4   32   3866.86

x(age)   y(expenses)
19	      16884.92


a and b is given
b=(n∑(xy) -∑(x)sum(y))/(n∑(x**2)- ∑(x)**2)
n= no. of data
∑(xy)
∑(x)
∑(y)
∑(x**2)
∑(x)**2

a=mean(y)-b*mean(x)

now
y=a + bx (now x ma jati value diyo tyo y ko value hunch)
